tokens/src/lib.rs

```diff
@@ -765,11 +751,11 @@ impl<T: Config> Pallet<T> {
 				// Ignore the result, because if it failed then there are remaining consumers,
 				// and the account storage in frame_system shouldn't be reaped.
 				let _ = frame_system::Pallet::<T>::dec_providers(who);
-				T::OnKilledTokenAccount::happened(&(who.clone(), currency_id));
+				<T::CurrencyHooks as CurrencyMutationHooks<T::AccountId, T::CurrencyId, T::Balance>>::OnKilledTokenAccount::happened(&(who.clone(), currency_id));
```

Comment:
you can define a `MutationHooksOf<T>` type alias to avoid repeating

Reply:
I'm not sure how that would work exactly? The error I'm getting with the type alias is
```rust
error[E0223]: ambiguous associated type
   --> tokens/src/lib.rs:941:3
    |
941 |         MutationHooksOf::<T>::TransferPostHook::on_transfer(
    |         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ help: use fully-qualified syntax: `<<T as module::Config>::CurrencyHooks as Trait>::TransferPostHook` 
```

Which happens because of [this issue](https://github.com/rust-lang/rust/issues/38078).